File: plugins/codemodder-plugin-maven/src/main/java/io/codemodder/plugins/maven/operator/SimpleUpgrade.java
package io.codemodder.plugins.maven.operator;

/** Represents bumping an existing dependency. */
class SimpleUpgrade extends AbstractCommand {

  private static SimpleUpgrade instance;

  private SimpleUpgrade() {}

  public static SimpleUpgrade getInstance() {
    if (instance == null) {
      instance = new SimpleUpgrade();
    }
    return instance;
  }

  @Override
  public boolean execute(ProjectModel pm) {
    if (pm.getDependency() == null) {
      throw new NullPointerException("Dependency must not be null.");
    }

    String lookupExpressionForDependency =
        Util.buildLookupExpressionForDependency(pm.getDependency());

    return handleDependency(pm, lookupExpressionForDependency);
  }
}

File: pom-operator/src/main/java/io/github/pixee/maven/operator/SimpleUpgrade.kt
package io.github.pixee.maven.operator

import io.github.pixee.maven.operator.Util.buildLookupExpressionForDependency


/**
 * Represents bumping an existing dependency/
 */
val SimpleUpgrade = object : AbstractCommand() {
    override fun execute(pm: ProjectModel): Boolean {
        val lookupExpressionForDependency =
            buildLookupExpressionForDependency(pm.dependency!!)

        return handleDependency(pm, lookupExpressionForDependency)
    }
}